core/rfb.js

```diff
@@ -167,13 +169,13 @@ export default class RFB extends EventTargetMixin {
 
         // Cursor
         this._cursor = new Cursor();
-        this._cursorImage = {
-            rgbaPixels: [],
-            x: 0,
-            y: 0,
-            w: 0,
-            h: 0,
-        };
+
+        // TightVNC 2.8.11 sends no cursor at all until Windows changes it, so
+        // let's show something visible by default. An arrow is what a user
+        // would expect, but it is too intrusive for servers that always draw
+        // the cursor in the remote framebuffer image. So let's show a small
+        // dot as a compromise.
+        this._cursorImage = RFB.cursors.dot;
```

Comment:
I'm not a fan of this. It would punish well behaving servers because of a bug in TightVNC. I think it is better that users will have to use this new setting for such buggy servers.

Reply:
Unfortunately the spec isn't clear on this point, so we'll have to go with observed behaviour. Servers that do server-side rendering of the cursor don't always send an empty local cursor (probably because it is an extension, i.e. optional).

Reply:
Given the unclear spec, we cannot call TightVNC behavior a bug. And I agree regarding the servers that do server-side rendering of the cursor. So definitely a case of conflicting requirements, OK, let's resolve it in favor of servers that do server-side rendering of the cursor and thus want the empty initial cursor.